manager.cpp

```diff
@@ -272,8 +277,104 @@ void Manager::updateObjects(
     }
 }
 
-void Manager::notify(std::map<sdbusplus::message::object_path, Object> objs)
+void Manager::clearPersistency(const vector<string>& paths)
+{
+    string systemPath =
+        "/var/lib/phosphor-inventory-manager/xyz/openbmc_project/inventory";
+
+    for (const auto& objPath : paths)
+    {
+        string absObjPath(systemPath);
+        absObjPath.append(objPath);
+
+        for (const auto& file : directory_iterator(absObjPath))
```

Comment:
I think you could just use std::filesystem::remove_all here to delete a directory and all of its contents.

Reply:
we don't want to delete all of it's content.
We have to delete all the files inside it,  and not any directory inside it.
After deleting all the files If target dir is empty (that means it had only files), then we delete the target dir too.

Reply:
ack